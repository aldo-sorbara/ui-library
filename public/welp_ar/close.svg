<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
