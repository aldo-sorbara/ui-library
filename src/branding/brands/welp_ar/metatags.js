export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
