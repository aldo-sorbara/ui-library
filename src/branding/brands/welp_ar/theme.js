import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
