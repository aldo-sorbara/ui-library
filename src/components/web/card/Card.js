import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

/**
 * @render react
 * @name Card
 * @description Card Component for Wenance
 * @example
 * <Card data="card"> This is a card container.</Card>
 */

const Card = props => {
  const { children, data, ...rest } = props;
  const Component = StyledCard;
  return (
    <Component {...rest} data-w={data}>
      {children}
    </Component>
  );
};

Card.propTypes = {
  data: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  active: PropTypes.bool
};

Card.defaultProps = {
  active: false
};

const StyledCard = styled.div`
  border: ${props => `1px solid ${props.theme.palette.common.disabled}`};
  box-shadow: ${props =>
    props.active ? '0 3px 14px 0 rgba(21, 67, 118, 0.12)' : 'none'};
  flex-direction: column;
  justify-content: center;
  margin-bottom: ${props => props.theme.spacer.medium};
  outline: none;
  padding: ${props => props.theme.spacer.medium};
`;

StyledCard.defaultProps = {
  theme: {
    spacer: {
      medium: '1rem'
    },
    palette: {
      common: {
        disabled: '#d5d8df'
      }
    }
  }
};

export default Card;
