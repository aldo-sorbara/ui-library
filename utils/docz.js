export const styles = {
  display: 'flex',
  flex: 1,
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-around'
};
