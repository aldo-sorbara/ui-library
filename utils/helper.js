import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
