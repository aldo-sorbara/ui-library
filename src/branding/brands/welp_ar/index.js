import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
