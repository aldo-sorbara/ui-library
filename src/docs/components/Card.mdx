---
name: Card
menu: Components
route: /card
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import Card from '../../components/web/card';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';
import { styles } from '../../../utils/docz';

# Card

<PropsTable of={Card} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <Card data='card'>This is a Card container.</Card>
      <Card active data='card'>
        This is an active Card container.
      </Card>
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/card--with-default-props)
