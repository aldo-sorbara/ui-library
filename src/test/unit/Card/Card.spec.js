import React from 'react';
import Card from '../../../components/web/card';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('Card component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Card className='test-card' data='card'>
          Card container
        </Card>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render an active card', () => {
    const wrapper = mountWithTheme(
      <Card className='test-card' active data='active-card'>
        Active Card
      </Card>,
      welp_ar.theme
    );
    expect(wrapper.props().active).toBe(true);
    expect(wrapper).toHaveStyleRule(
      'box-shadow',
      '0 3px 14px 0 rgba(21,67,118,0.12)'
    );
    expect(wrapper).toHaveStyleRule('border', '1px solid #d5d8df');
  });
  it('should render card with default props', () => {
    const wrapper = mountWithTheme(
      <Card className='test-card' data='default-card'>
        Card default
      </Card>,
      welp_ar.theme
    );
    expect(wrapper.props().active).toBe(false);

    expect(wrapper).toHaveStyleRule('border', '1px solid #d5d8df');
    expect(wrapper).toHaveStyleRule('box-shadow', 'none');
    expect(wrapper).toHaveStyleRule('padding', '1rem');
  });
});
