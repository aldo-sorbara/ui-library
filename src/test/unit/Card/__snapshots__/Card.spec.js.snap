// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render snapshot 1`] = `
.c0 {
  border: 1px solid #d5d8df;
  box-shadow: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 1rem;
  outline: none;
  padding: 1rem;
}

<div
  className="test-card c0"
  data-w="card"
>
  Card container
</div>
`;
